go/enclave/components/rollup_consumer.go

```diff
@@ -123,55 +122,62 @@ func (rc *rollupConsumerImpl) processRollups(br *common.BlockAndReceipts) ([]*co
 		return nil, nil
 	}
 
+	validRollups := make([]*core.Rollup, 0)
+
 	blockHash := block.Hash()
-	for idx, rollup := range rollups {
+	for _, rollup := range rollups {
```

Comment:
I'm not sure about this logic to handle multiple rollups in a block.

I would say we add a check that there can be a single rollup signed the sequencer in a block.

This should simplify things, as you can assume there is a single one, and it has to be on top of the previous one.

Reply:
We'd need to modify the smart contract to also disallow any other rollup submissions in the same block for this to work. Otherwise someone else could resubmit old rollups and hide new ones.

Reply:
I've used the assumption of one signed rollup per block and put a 'todo' to make sure we get rid of the DoS potential.